### PR TITLE
qsh verification 

### PR DESCRIPTION
**Pull request by skv-headless**
copying behaviour from atlassian-connect-express https://bitbucket.org/atlassian/atlassian-connect-express/src/58263a23be29b5603367f47341e984725b6a89b6/lib/middleware/authentication.js#lines-217

based on
https://community.developer.atlassian.com/t/action-required-atlassian-connect-vulnerability-allows-bypass-of-app-qsh-verification-via-context-jwts/47072

Probably I need to put into README explanation of what is qsh?

### CODE REVIEW

**Comment by pawelniewie** [comment]
Please update the version too.

**Comment by pawelniewie** [comment]
Yes, please update README and explain when you must verify qsh and in what cases you want to skip it.

File: lib/atlassian-jwt-authentication/verify.rb [hunk 1]
```diff
@@ -81,12 +81,7 @@ def verify
 
         qsh = Digest::SHA256.hexdigest(qsh)
 
-        unless data['qsh'] == qsh
-          log(:error, "QSH mismatch for client_key #{data['iss']} and addon_key #{addon_key}")
```
Comment (pawelniewie) [inline lib/atlassian-jwt-authentication/verify.rb:85]: @bulinutza did this log line turned useful to you? Do you depend on it or can it be removed?

**Comment by bulinutza** [comment]
We never had verification fail because of qsh mismatch, so I'm ok with
removing it.

**Comment by skv-headless** [comment]
- [x] copied section **How to send a signed HTTP request from the iframe back to the add-on service** from ACE. ACE docs don't explain what is qsh so I guess we should not too. 
- [x] Bumped new version

**Review by pawelniewie — APPROVED** [review]

**Comment by pawelniewie** [comment]
@bulinutza you wanna take a look at this or should I merge it?

**Pull request merged**